pkg/metrics/job_metrics.go

```diff
@@ -154,16 +167,21 @@ func (m *JobMetrics) FirstPodLaunchDelaySeconds(job metav1.Object, status v1.Job
 }
 
 func (m *JobMetrics) AllPodsLaunchDelaySeconds(pods []*corev1.Pod, job metav1.Object, status v1.JobStatus) {
-	if status.StartTime == nil {
+	if !util.IsRunning(status) || status.StartTime == nil {
```

Comment:
Similarly, the StartTime definition is different for xdl,xgboost jobs.  we may use job's creationTimestamp